### PR TITLE
refactor query and option compilation

### PR DESCRIPTION
**Pull request by walterddr**
This is part2 follow up of #9191 

- move option parsing and query compile to common utils
- support option string parsing as well from controller rest api

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/SqlNodeAndOptions.java
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/SqlNodeAndOptions.java:31]: debug options will be removed once release 0.11 is cut. so we will wait until then before merging this PR.

**Review by siddharthteotia — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:141]: The new added `sql` seems unused?

File: pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java:71]: This is not enough. During version upgrade, server doesn't know the debug options can be read from the query options

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java:137]: This needs to be moved into the try-catch

**Review by Jackie-Jiang — COMMENTED** [review]
We cannot merge this PR before 0.11 is cut. Currently it is still in release candidate branch, so I'd suggest reverting it for now for safety reason. Once `0.11` is officially released, we can re-post the commit

**Pull request merged**
